src/store/api/contentful.ts

```diff
@@ -0,0 +1,81 @@
+import { createClient, type EntryFieldTypes, type Entry } from 'contentful';
+import { createApi } from '@reduxjs/toolkit/query/react';
+
+type FromSkeleton<T extends ContentfulSkeletons> = Entry<T, 'WITHOUT_UNRESOLVABLE_LINKS'>['fields'];
+
+type AdSkeleton = {
+  contentTypeId: 'ads',
+  fields: {
+    title: EntryFieldTypes.Text,
+    description: EntryFieldTypes.Text,
+    fullscreen: EntryFieldTypes.Boolean,
+    poster: EntryFieldTypes.AssetLink
+  }
+};
+type Ad = FromSkeleton<AdSkeleton>;
+
+type BoardMessageSkeleton = {
+  contentTypeId: 'board-message',
+  fields: {
+    message: EntryFieldTypes.Text
+  }
+};
+type BoardMessage = FromSkeleton<BoardMessageSkeleton>;
+
+type QuoteSkeleton = {
+  contentTypeId: 'quotes',
+  fields: {
+    person: EntryFieldTypes.Text,
+    text: EntryFieldTypes.Text
+  }
+};
+type Quote = FromSkeleton<QuoteSkeleton>;
+
+type ContentfulSkeletons = AdSkeleton | BoardMessageSkeleton | QuoteSkeleton;
+
+type ContentfulResponse<ContentType extends ContentfulSkeletons["contentTypeId"]> =
```

Comment:
What you are doing here is quite some typescript magic, which makes the code more difficult for beginners. 

I think (/ expect) that use simpler generics, the content is way more easier to understand